File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmB5Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import java.time.Duration;
import java.util.List;
import java.util.UUID;
import systems.zlink.e2e.spotservice.shared.Contracts;
import systems.zlink.stream.connector.ZLinkStreamConnector;

public final class SmB5Scenario extends SpotServiceScenarioContext {
    private SmB5Scenario(SpotServiceScenarioContext context) {
        super(context);
    }

    public static void run(SpotServiceScenarioContext context) {
        new SmB5Scenario(context).execute();
    }

    private void execute() {
        String actorId = "actor-sm-b5-missing-" + UUID.randomUUID().toString().replace("-", "");
        ZLinkStreamConnector connector = createStreamConnector(options().streamAEndpoint());
        Contracts.ActorProfile profile = new Contracts.ActorProfile("Missing", 5, List.of("missing"));
        try {
            connector.connect().submit().toCompletableFuture().join();
            Contracts.ActorAuthRes auth = connector
                .request(new Contracts.ActorAuthReq(actorId, profile))
                .submit(Contracts.ActorAuthRes.class).toCompletableFuture().join();
            ensure(actorId.equals(auth.actorId()), "SM-B5 auth actor mismatch");

            expectFailure(() -> {
                try {
                    connector
                        .request(new Contracts.MissingActorReq("missing-handler"))
                        .metadata("actor-id", actorId)
                        .timeout(Duration.ofSeconds(2))
                        .submit(Contracts.ActorEchoRes.class).toCompletableFuture().join();
                } catch (Exception error) {
                    throw new RuntimeException(error);
                }
            });

            Contracts.EvidenceSnapshot evidence = waitForPlayAEvidence(List.of(
                "HANDLER_MISSING/REPLY_ERROR/MissingActorReq"));
            ensure(evidence.entries().stream().anyMatch(entry ->
                    "DispatchError".equals(entry.marker())
                        && "HANDLER_MISSING/REPLY_ERROR/MissingActorReq".equals(entry.value())),
                "SM-B5 missing actor dispatch error evidence mismatch");

            System.out.println("scenario SM-B5 passed");
        } catch (Exception error) {
            throw new IllegalStateException("actor missing handler scenario failed", error);
        } finally {
            closeQuietly(connector);
        }

    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmB5Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.framework.kotlin.*

import java.time.Duration
import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.Env
import systems.zlink.e2e.kotlin.spotservice.client.support.createStreamConnector
import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.expectFailure
import systems.zlink.e2e.kotlin.spotservice.client.support.postJson

internal object SmB5Scenario {
    suspend fun run() {
        val connector = createStreamConnector(Env.get("e2e.stream.a.endpoint"))
        try {
            val actorId = "actor-sm-b5-missing"
            val profile = Contracts.ActorProfile("Missing Handler", 5, listOf("missing"))
            connector.connect().await()
            val auth = connector
                .request(Contracts.ActorAuthReq(actorId, profile))
                .awaitReply<Contracts.ActorAuthRes>()
            ensure(auth.actorId == actorId, "SM-B5 auth actor mismatch")

            expectFailure {
                connector
                    .request(Contracts.MissingActorReq("missing-handler", 5, profile))
                    .metadata("actor-id", actorId)
                    .timeout(Duration.ofSeconds(2))
                    .awaitReply<Contracts.ActorEchoRes>()
            }

            postJson(
                Env.get("e2e.http.session.endpoint"),
                "/evidence/wait",
                Contracts.EvidenceWaitReq(
                    listOf("SPOT_ACTOR|HANDLER_MISSING/REPLY_ERROR/MissingActorReq"),
                    10_000,
                ),
                Contracts.EvidenceSnapshot::class.java,
            )

            println("scenario SM-B5 passed")
        } finally {
            try {
                connector.close().await()
            } catch (_: Exception) {
            }
        }
    }
}
